test/libsolidity/syntaxTests/dataLocations/variable_declaration_location_transient_test_reference_type.sol

```diff
@@ -0,0 +1,14 @@
+contract test {
+    struct S {
+        uint x;
+    }
+    S transient b;
+
+    uint[] transient a;
+    function f() public {
+        uint[] transient c;
+        S transient d;
+    }
+}
+// ----
+// ParserError 2314: (148-149): Expected ';' but got identifier
```

Comment:
At first, I had implemented `transient` being accepted by the parser as data location for any variable and the during type checking only allow it for state vars. Then later, this was changed to only allowing parsing of `transient` as data location for state vars. So, in the beginning this test made more sense (also it is reflecting an already existing test (`test/libsolidity/syntaxTests/dataLocations/variable_declaration_location_specifier_test_non_reference_type.sol`)...now, there are other tests which better cover the cases here (reference type state vars and reference type local vars). I think that then we could exclude this, or at least, make it only about state vars...